pkg/genericcli/context.go

```diff
@@ -0,0 +1,682 @@
+			cmd.Flags().Bool(KeyActivate, false, "immediately switches to the new context")
+			cmd.Flags().String(KeyProvider, "", "set the login provider for this context")
+
+			Must(cmd.MarkFlagRequired(KeyName))
+			Must(cmd.MarkFlagRequired(KeyAPIToken))
+
+			cmd.Args = cobra.ExactArgs(0)
+		},
+		UpdateCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().String(KeyAPIURL, "", "set the api-url for this context")
+			cmd.Flags().String(KeyAPIToken, "", "set the api-token for this context")
+			cmd.Flags().String(KeyDefaultProject, "", "set a default project to operate on")
+			cmd.Flags().Duration(KeyTimeout, 0, "set a default request timeout")
+			cmd.Flags().Bool(KeyActivate, false, "immediately switches to the new context")
+			cmd.Flags().String(KeyProvider, "", "set the login provider for this context")
+
+			Must(cmd.RegisterFlagCompletionFunc(KeyDefaultProject, c.ProjectListCompletion))
+
+			cmd.ValidArgsFunction = c.ContextListCompletion
+
+			cmd.Args = cobra.ExactArgs(1)
+		},
+		DeleteCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.ValidArgsFunction = c.ContextListCompletion
+
+			cmd.Args = cobra.ExactArgs(1)
+		},
+		CreateRequestFromCLI: func() (*Context, error) {
+			return &Context{}, nil // Placeholder to trigger cmdline read (not file read)
+		},
+		UpdateRequestFromCLI: func(args []string) (*contextUpdateRequest, error) {
+			name, err := GetExactlyOneArg(args)
+			if err != nil {
+				return nil, fmt.Errorf("no context name given")
+			}
+			return &contextUpdateRequest{Name: name}, nil
+		},
+	})
+
+	switchCmd := &cobra.Command{
+		Use:     "switch <context-name>",
+		Short:   "switch the active CLI context",
+		Long:    "Switch the active CLI context. Use \"-\" to switch to the previously used context.",
+		Aliases: []string{"set", "sw"},
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.setContext(args)
+		},
+		ValidArgsFunction: c.ContextListCompletion,
+	}
+
+	setProjectCmd := &cobra.Command{
+		Use:   "set-project <project-id>",
+		Args:  cobra.ExactArgs(1),
+		Short: "set the default project to operate on for cli commands",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.setProject(args)
+		},
+		ValidArgsFunction: c.ProjectListCompletion,
+	}
+
+	showCurrentCmd := &cobra.Command{
+		Use:   "show-current",
+		Short: "print the active context name",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctxs, err := c.GetContexts()
+			if err != nil {
+				return fmt.Errorf("unable to get contexts: %w", err)
+			}
+			if ctxs.CurrentContext == "" {
+				return fmt.Errorf("no context currently active")
+			}
+
+			_, err = fmt.Fprint(c.Out, ctxs.CurrentContext)
+			return err
+		},
+		ValidArgsFunction: c.ContextListCompletion,
+	}
+
+	cmd.AddCommand(
+		switchCmd,
+		setProjectCmd,
+		showCurrentCmd,
+	)
+
+	return cmd
+}
+
+func (c *ContextConfig) setContext(args []string) error {
```

Comment:
`ContextConfig` should not have additional functions, everything regarding the management of the contexts should be on the same struct that implements the CRUD interface.